### PR TITLE
slack-status: Added support for pausing notifications

### PR DESCRIPTION
**Pull request by henrik**
## Description

The Slack Status extension gets a "Pause notifications" checkbox for statuses. If checked, Slack notifications are paused for the duration of this status.

If the status is set to "Don't clear", notifications are paused for 12 hours – they can't be paused indefinitely via API.

The list showing the current status and the status presets also gets a 🔕 accessory indicating that notifications are paused.

Old presets.json silently get migrated.

Because new Slack permissions are required, a user without those permissions (i.e. upgraders as opposed to fresh installers) will be shown a confirmation dialogue and then required to reconnect to Slack.

## Screenshots

Screenshot 2024-12-14 at 23 37 12

Screenshot 2024-12-14 at 23 37 17

Screenshot 2024-12-14 at 23 39 01

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/slack-status/src/actions.tsx [hunk 1]
```diff
@@ -369,16 +342,18 @@ export function EditStatusPresetAction(props: {
       target={
         <StatusForm
           actionTitle="Update Preset"
-          initalValues={{
+          initialValues={{
```
Comment (henrik) [inline extensions/slack-status/src/actions.tsx:345]: I snuck in some typo fixing…

File: extensions/slack-status/src/actions.tsx [hunk 2]
```diff
@@ -70,7 +72,7 @@ export function SetStatusWithAIAction(props: { statusText: string; mutate: Mutat
               `You help a Slack user set their status.
               
               **Respond with a JSON object with the following attributes:**
-              - "text": a string value for status text, should be short and sweet, with no punctuation, e.g. "Working out", "Listening to Drake's new album", "Coffe break". It should not include the status duration for example "Working out" instead of "Working out for 2 hours" or "Working out until tomorrow".
+              - "text": a string value for status text, should be short and sweet, with no punctuation, e.g. "Working out", "Listening to Drake's new album", "Coffee break". It should not include the status duration for example "Working out" instead of "Working out for 2 hours" or "Working out until tomorrow".
```
Comment (henrik) [inline extensions/slack-status/src/actions.tsx:75]: Another snuck-in typo fix.

I considered updating the AI action to be able to set DND, but it seemed unlikely to be used and would add complexity, so I skipped it.

File: extensions/slack-status/src/slack.ts [hunk 3]
```diff
@@ -9,13 +10,45 @@ export function useSlack() {
...
+
+    while (!dndCalled) {
+      try {
+        dndResponse = await slack.dnd.info();
+        dndCalled = true;
+        if (!dndResponse.ok) {
+          throw Error("Failed to fetch DND info");
+        }
+      } catch (error) {
+        // We added new scopes for DND. Old installs don't automatically upgrade scopes.
```
Comment (henrik) [inline extensions/slack-status/src/slack.ts:33]: As discussed in https://raycastcommunity.slack.com/archives/C02HEMAF2SJ/p1734186555949399.

File: extensions/slack-status/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # Slack Status Change
 
+## [Support pausing notifications] - 2024-12-14
```
Comment (henrik) [inline extensions/slack-status/CHANGELOG.md:3]:
```suggestion
## [Support pausing notifications] - {PR_MERGE_DATE}
```

**Comment by henrik** [comment]
I ran `npm run build` and TypeScript is angry with me. I'll see if I can figure it out.

**Comment by henrik** [comment]
>  I checked that files in the assets folder are used by the extension itself

slack.svg appears unused, but that is unrelated to this PR. It was that way before.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by henrik** [comment]
Thank you! ❤️ 😊

**Pull request merged**